mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -2896,6 +2896,114 @@ LogicalResult TransposeConv2DOp::inferReturnTypeComponents(
 LogicalResult TransposeConv2DOp::verify() {
   if (verifyConvOp(*this).failed() || verifyConvOpModes(*this).failed())
     return failure();
+
+  llvm::ArrayRef<int64_t> strides = getStride();
+  const int64_t strideY = strides[0];
+  const int64_t strideX = strides[1];
+
+  if (strideY < 1 || strideX < 1)
+    return emitOpError("expect all stride values to be >= 1, got [")
+           << strides << "]";
+
+  const RankedTensorType inputType =
+      llvm::dyn_cast<RankedTensorType>(getInput().getType());
+
+  const RankedTensorType outputType =
+      llvm::dyn_cast<RankedTensorType>(getOutput().getType());
+
+  const RankedTensorType weightType =
+      llvm::dyn_cast<RankedTensorType>(getWeight().getType());
+
+  llvm::ArrayRef<int64_t> padding = getOutPad();
+
+  const auto checkPadAgainstKernelDim =
+      [this](const int64_t pad_value, const int64_t kernel_dim_size,
```

Comment:
Is there a particular reason for passing these arguments by const value as opposed to just by value?

Reply:
Good spot, I don't think the const makes any difference in this case, removed it.